system/Cookie/Cookie.php

```diff
@@ -37,6 +37,8 @@
  * $cookie2 = $cookie->withName('prod_cookie');
  * $cookie2->getName(); // prod_cookie
  * ```
+ *
+ * @template-implements ArrayAccess<string, bool|int|string>
```

Comment:
This is generic typing. If we want to impose this too for phpstan, we may consider raising phpstan level to 6.

Reply:
5 to 6 is hard for existing code bases. I would of course love to see our level increase, so maybe we can baseline existing array issues and make the move.